airflow/contrib/operators/ssh_operator.py

```diff
@@ -96,15 +101,14 @@ def execute(self, context):
 
             with self.ssh_hook.get_conn() as ssh_client:
                 # Auto apply tty when its required in case of sudo
-                get_pty = False
                 if self.command.startswith('sudo'):
-                    get_pty = True
+                    self.get_pty = True
```

Comment:
I think it might make sense to move this to constructor. command is set only in the constructor so we can keep all the pty logic there and it could be one-liner 
`self.get_pty = self.start_command.starstwith('sudo') or self.get_pty`